### ISSUE TITLE
Installing `mirage-dns` with `--build-test` fails

### ISSUE THREAD

**Issue by bbc2** [mirage/ocaml-dns#133]
Hello, here's the output I got from opam. I don't really understand why it fails. Perhaps `--build-test` is not supposed to be supported?

```
> opam install --build-test mirage-dns
The following actions will be performed:
  ∗  install conf-m4          1               [required by ocamlfind]
  ∗  install result           1.2             [required by dns]
  ∗  install ocamlbuild       0.11.0          [required by dns]
  ∗  install conf-which       1               [required by hashcons]
  ∗  install ocamlfind        1.7.1           [required by mirage-dns]
  ∗  install uchar            0.0.1           [required by fmt]
  ∗  install conf-autoconf    0.1             [required by hashcons]
  ∗  install topkg            0.9.0           [required by dns]
  ∗  install sexplib          113.33.00+4.03  [required by ipaddr, cstruct, uri]
  ∗  install ppx_tools        5.0             [required by dns]
  ∗  install ounit            2.0.0           [required by dns]
  ∗  install base-bytes       base            [required by dns]
  ∗  install hashcons         1.2             [required by dns]
  ∗  install rresult          0.5.0           [required by base64]
  ∗  install mtime            0.8.4           [required by bos]
  ∗  install cmdliner         1.0.0           [required by dns]
  ∗  install astring          0.8.3           [required by alcotest, bos]
  ∗  install ppx_core         113.33.01+4.03  [required by ppx_sexp_conv]
  ∗  install lwt              2.7.0           [required by mirage-dns]
  ∗  install re               1.7.1           [required by dns]
  ∗  install qcheck           0.5.3.1         [required by qtest]
  ∗  install cppo             1.4.1           [required by ocplib-endian]
  ∗  install fmt              0.8.2           [required by mirage-kv, mirage-protocols]
  ∗  install fpath            0.7.1           [required by bos]
  ∗  install ppx_optcomp      113.33.01+4.03  [required by ppx_driver]
  ∗  install qtest            2.5             [required by stringext]
  ∗  install ppx_deriving     4.1             [required by ppx_type_conv]
  ∗  install ocplib-endian    1.0             [required by cstruct, mirage-profile]
  ∗  install mirage-flow      1.2.0           [required by mirage-protocols]
  ∗  install mirage-device    1.0.0           [required by mirage-kv, mirage-protocols]
  ∗  install logs             0.6.2           [required by bos]
  ∗  install alcotest         0.7.2           [required by base64, duration]
  ∗  install ppx_driver       113.33.02+4.03  [required by ppx_type_conv]
  ∗  install stringext        1.4.3           [required by uri]
  ∗  install cstruct          2.3.2           [required by dns]
  ∗  install mirage-time      1.0.0           [required by mirage-time-lwt]
  ∗  install mirage-protocols 1.1.0           [required by dns]
  ∗  install mirage-kv        1.0.0           [required by dns]
  ∗  install bos              0.1.5           [required by base64]
  ∗  install duration         0.1.0           [required by mirage-dns]
  ∗  install ppx_type_conv    113.33.02+4.03  [required by ppx_sexp_conv]
  ∗  install pcap-format      0.5.0           [required by dns]
  ∗  install io-page          1.6.1           [required by mirage-dns]
  ∗  install mirage-time-lwt  1.0.0           [required by mirage-dns]
  ∗  install mirage-stack     1.0.0           [required by mirage-stack-lwt]
  ∗  install mirage-kv-lwt    1.0.0           [required by mirage-dns]
  ∗  install base64           2.1.2           [required by dns]
  ∗  install ppx_sexp_conv    113.33.01+4.03  [required by ipaddr, uri]
  ∗  install mirage-profile   0.7.0           [required by mirage-dns]
  ∗  install uri              1.9.2           [required by dns]
  ∗  install ipaddr           2.7.2           [required by dns]
  ∗  install mirage-stack-lwt 1.0.0           [required by mirage-dns]
  ∗  install dns              0.19.1          [required by mirage-dns]
  ∗  install mirage-dns       2.6.1         
===== ∗  54 =====
Do you want to continue ? [Y/n] y

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[alcotest] Archive in cache
[astring] Archive in cache
[base64] Archive in cache
[bos] Archive in cache
[cmdliner] Archive in cache
[cppo] Archive in cache
[cstruct] Archive in cache
[dns] Archive in cache
[duration] Archive in cache
[fmt] Archive in cache
[fpath] Archive in cache
[hashcons] Archive in cache
[io-page] Archive in cache
[logs] Archive in cache
[lwt] Archive in cache
[mirage-device] Archive in cache
[mirage-dns] Archive in cache
[mirage-flow] Archive in cache
[mirage-kv] Archive in cache
[mirage-kv-lwt] Archive in cache
[mirage-profile] Archive in cache
[mirage-protocols] Archive in cache
[mirage-stack] Archive in cache
[mirage-stack-lwt] Archive in cache
[mirage-time] Archive in cache
[mirage-time-lwt] Archive in cache
[mtime] Archive in cache
[ocamlbuild] Archive in cache
[ocamlfind] Archive in cache
[ocplib-endian] Archive in cache
[ounit] Archive in cache
[pcap-format] Archive in cache
[ppx_core] Archive in cache
[ppx_deriving] Archive in cache
[ppx_driver] Archive in cache
[ppx_optcomp] Archive in cache
[ppx_sexp_conv] Archive in cache
[ppx_tools] Archive in cache
[ppx_type_conv] Archive in cache
[qcheck] Archive in cache
[qtest] Archive in cache
[re] Archive in cache
[result] Archive in cache
[rresult] Archive in cache
[sexplib] Archive in cache
[stringext] Archive in cache
[topkg] Archive in cache
[uchar] Archive in cache
[uri] Archive in cache
[default] https://opam.ocaml.org/archives/ipaddr.2.7.2+opam.tar.gz downloaded

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
∗  installed conf-m4.1
∗  installed conf-which.1
∗  installed conf-autoconf.0.1
∗  installed result.1.2
∗  installed ocamlfind.1.7.1
∗  installed base-bytes.base
∗  installed hashcons.1.2
∗  installed ppx_tools.5.0
∗  installed ocamlbuild.0.11.0
∗  installed cppo.1.4.1
∗  installed ocplib-endian.1.0
∗  installed ounit.2.0.0
∗  installed sexplib.113.33.00+4.03
∗  installed ppx_deriving.4.1
∗  installed ppx_core.113.33.01+4.03
∗  installed ppx_optcomp.113.33.01+4.03
∗  installed qcheck.0.5.3.1
∗  installed ppx_driver.113.33.02+4.03
∗  installed qtest.2.5
∗  installed lwt.2.7.0
∗  installed ppx_type_conv.113.33.02+4.03
∗  installed re.1.7.1
∗  installed ppx_sexp_conv.113.33.01+4.03
∗  installed uchar.0.0.1
∗  installed stringext.1.4.3
∗  installed topkg.0.9.0
∗  installed astring.0.8.3
∗  installed cmdliner.1.0.0
∗  installed fpath.0.7.1
∗  installed fmt.0.8.2
∗  installed alcotest.0.7.2
∗  installed duration.0.1.0
∗  installed ipaddr.2.7.2
∗  installed mirage-device.1.0.0
∗  installed mirage-flow.1.2.0
∗  installed mirage-kv.1.0.0
∗  installed mirage-time.1.0.0
∗  installed mirage-protocols.1.1.0
∗  installed mirage-time-lwt.1.0.0
∗  installed mirage-stack.1.0.0
∗  installed rresult.0.5.0
∗  installed uri.1.9.2
∗  installed mtime.0.8.4
∗  installed logs.0.6.2
∗  installed cstruct.2.3.2
∗  installed mirage-kv-lwt.1.0.0
∗  installed mirage-stack-lwt.1.0.0
∗  installed bos.0.1.5
∗  installed pcap-format.0.5.0
∗  installed base64.2.1.2
∗  installed io-page.1.6.1
∗  installed mirage-profile.0.7.0
∗  installed dns.0.19.1
[ERROR] The compilation of mirage-dns failed at "ocamlfind query dns.mirage".

#=== ERROR while installing mirage-dns.2.6.1 ==================================#
# opam-version 1.2.2
# os           linux
# command      ocamlfind query dns.mirage
# path         /home/bertrand/.opam/mirage-issue/build/mirage-dns.2.6.1
# compiler     4.04.0
# exit-code    2
# env-file     /home/bertrand/.opam/mirage-issue/build/mirage-dns.2.6.1/mirage-dns-3336-cd16b6.env
# stdout-file  /home/bertrand/.opam/mirage-issue/build/mirage-dns.2.6.1/mirage-dns-3336-cd16b6.out
# stderr-file  /home/bertrand/.opam/mirage-issue/build/mirage-dns.2.6.1/mirage-dns-3336-cd16b6.err
### stderr ###
# ocamlfind: Package `dns.mirage' not found



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions failed
  ∗  install mirage-dns 2.6.1
The following changes have been performed
  ∗  install alcotest         0.7.2         
  ∗  install astring          0.8.3         
  ∗  install base-bytes       base          
  ∗  install base64           2.1.2         
  ∗  install bos              0.1.5         
  ∗  install cmdliner         1.0.0         
  ∗  install conf-autoconf    0.1           
  ∗  install conf-m4          1             
  ∗  install conf-which       1             
  ∗  install cppo             1.4.1         
  ∗  install cstruct          2.3.2         
  ∗  install dns              0.19.1        
  ∗  install duration         0.1.0         
  ∗  install fmt              0.8.2         
  ∗  install fpath            0.7.1         
  ∗  install hashcons         1.2           
  ∗  install io-page          1.6.1         
  ∗  install ipaddr           2.7.2         
  ∗  install logs             0.6.2         
  ∗  install lwt              2.7.0         
  ∗  install mirage-device    1.0.0         
  ∗  install mirage-flow      1.2.0         
  ∗  install mirage-kv        1.0.0         
  ∗  install mirage-kv-lwt    1.0.0         
  ∗  install mirage-profile   0.7.0         
  ∗  install mirage-protocols 1.1.0         
  ∗  install mirage-stack     1.0.0         
  ∗  install mirage-stack-lwt 1.0.0         
  ∗  install mirage-time      1.0.0         
  ∗  install mirage-time-lwt  1.0.0         
  ∗  install mtime            0.8.4         
  ∗  install ocamlbuild       0.11.0        
  ∗  install ocamlfind        1.7.1         
  ∗  install ocplib-endian    1.0           
  ∗  install ounit            2.0.0         
  ∗  install pcap-format      0.5.0         
  ∗  install ppx_core         113.33.01+4.03
  ∗  install ppx_deriving     4.1           
  ∗  install ppx_driver       113.33.02+4.03
  ∗  install ppx_optcomp      113.33.01+4.03
  ∗  install ppx_sexp_conv    113.33.01+4.03
  ∗  install ppx_tools        5.0           
  ∗  install ppx_type_conv    113.33.02+4.03
  ∗  install qcheck           0.5.3.1       
  ∗  install qtest            2.5           
  ∗  install re               1.7.1         
  ∗  install result           1.2           
  ∗  install rresult          0.5.0         
  ∗  install sexplib          113.33.00+4.03
  ∗  install stringext        1.4.3         
  ∗  install topkg            0.9.0         
  ∗  install uchar            0.0.1         
  ∗  install uri              1.9.2         

=-=- lwt.2.7.0 installed successfully -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
=> The future Lwt 3.0.0 will make minor breaking changes near 1 April 2017. See
     https://github.com/ocsigen/lwt/issues/308

=-=- qtest.2.5 installed successfully -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
=> Version similar to 2.4, except it uses qcheck as an external library
     instead of embedding it. There should be no other difference.

     To use it, still call `qtest` on your .ml/.mli files to extract tests,
     then compile the resulting suite using oUnit and qcheck.

The former state can be restored with:
    opam switch import "~/.opam/mirage-issue/backup/state-20170221165342.export"
```

**Comment by avsm** [comment]
this is comprehensively fixed by the upcoming dns 1.0, which removes all optional dependencies and so dns-mirage will be a much more reliable installation

**Comment by bbc2** [comment]
This indeed doesn't seem to be an issue anymore.   Thanks!
